### ISSUE TITLE
Put high cardinality in different measurement or tags?

### ISSUE THREAD

**Issue by joezhoujinjing** [influxdata/influxdb#8003]
### Directions

A question:
We have many users with unique ids, should we the data for each id into different measurements? It seems putting them into tag is definitely not a good idea.

**Comment by jwilder** [comment]
High cardinality attributes should go into fields.  Putting them in the measurement still creates a high cardinality problem.

**Comment by mlosev** [comment]
> High cardinality attributes should go into fields. 

What if I want to query data by specific field value? Will it be fast?
Let's suppose I have 150000 unique user ids - this is larger than default limit per tag.
If I put user id in field and then want to query info about specific user id - will work as fast as with tags?
How should I structure my data to get fast querying of per user data?

**Comment by jwilder** [comment]
You can increase (or disable the limit).
```
[data]
max-values-per-tag = 0 # to disable
```

The limit is there to prevent run away high cardinality data from getting the server into a state where it crashes or is difficult to recover from.  If you have a lot of RAM, you can support higher cardinality.

Querying against data in a field won't be as fast as using a tag as the data is not indexed.  Whether it is fast enough would depend on your requirements and environment.

**Comment by logzilla** [comment]
> If you have a lot of RAM, you can support higher cardinality.
Is there some documentation that specifies what *a lot of ram* is?

I have a server using only 5GB ram with 60GB free. We have two tags that go to around 300k values. 

Given this, is there a way to calculate the expected amount of ram needed?

**Comment by logzilla** [comment]
Well, I found out myself. The truth is: it's not *just* about memory. I enabled the tags and within a few hours on a larger server, I had `129181` values. Influx was using about `10GB` more for the ram. This server has `227GB` ram free, yet....Influx's response came to a horrible screeching halt. It became more or less unusable, unable to make queries within a reasonable time frame.

I don't see why, with 200+ GB of ram free, that influx would degrade so terribly.
